src/audio/tdfb/tdfb.c

```diff
@@ -117,14 +124,39 @@ static void tdfb_free_delaylines(struct tdfb_comp_data *cd)
 		fir[i].delay = NULL;
 }
 
+static void *tdfb_filter_seek(struct sof_tdfb_config *config, int num_filters)
+{
+	struct sof_fir_coef_data *coef_data;
+	int i;
+	uint8_t *coefp = (uint8_t *)ASSUME_ALIGNED(&config->data[0], 2); /* 2 for 16 bits data */
```

Comment:
lets just call this a void *, as the uint8 is confusing (as it looks like its for data processing)

Reply:
This will also save on casting the return value and the alignment assumption.

Reply:
I tried but I get error "warning: pointer of type ‘void *’ used in arithmetic" for line

`coefp += sizeof(*coef_data) + sizeof(int16_t) * coef_data->length;`

Do you have proposals how to do it?

Reply:
The code is telling the CC the data is 16bit (aligned 2) so you could make coefp a uint16

Reply:
`config->data` is `int16_t` so let's make `int16_t *coefp;` and drop the type-cast